pkg/virt-handler/rest/console.go

```diff
@@ -153,10 +153,47 @@ func (t *ConsoleHandler) VNCHandler(request *restful.Request, response *restful.
 	}
 	uid := vmi.GetUID()
 	stopChn := newStopChan(uid, t.vncLock, t.vncStopChans)
-	defer deleteStopChan(uid, stopChn, t.vncLock, t.vncStopChans)
+
+	t.handleConcurrentConnectionStart(vmi, stopChn)
+	defer t.handleConcurrentConnectionEnd(vmi, stopChn)
+
 	t.stream(vmi, request, response, unixSocketDialer(vmi, unixSocketPath), stopChn)
 }
 
+
+var vncMap sync.Map
+
+// make concurrent vnc connection available
+func (t *ConsoleHandler) handleConcurrentConnectionStart(vmi *v1.VirtualMachineInstance, stopChn chan struct{}) {
+	uid := vmi.GetUID()
+
+	var currentCount int
+	if n, ok := vncMap.Load(uid); ok {
+		currentCount = n.(int)
+	}
+	newCount := currentCount + 1
+	log.Log.Object(vmi).Infof("VNC connection count: %d, newCount: %d", currentCount, newCount)
+	vncMap.Store(uid, newCount)
+}
+
+func (t *ConsoleHandler) handleConcurrentConnectionEnd(vmi *v1.VirtualMachineInstance, stopChn chan struct{}) {
+	uid := vmi.GetUID()
+
+	var currentCount int
+	if n, ok := vncMap.Load(uid); ok {
+		currentCount = n.(int)
+	}
+	newCount := currentCount - 1
+	log.Log.Object(vmi).Infof("VNC connection count: %d, newCount: %d", currentCount, newCount)
+	vncMap.Store(uid, newCount)
+
+	if newCount <= 0 {
```

Comment:
Comparison looks wrong

Reply:
I don't get it,  what's the comparison you'd expect, is it `newCount  < 0` ?